bindgen/ir/context.rs

```diff
@@ -2971,3 +2969,19 @@ impl TemplateParameters for PartialType {
         }
     }
 }
+
+fn unused_regex_diagnostic(item: &str, name: &str, emit_diagnostics: bool) {
```

Comment:
@pvdrz  Why do we have to add a special bool in this function as argument? why can't this function itself get `let emit_diagnostics = .....options().emit_diagnostics;`. Is it because we cannot access this option in that function? hmm....

Reply:
yeah I am wary of getting a reference to `ctx` as it might cause borrowing issues in some cases.

Reply:
makes sense.